mne/source_estimate.py

```diff
@@ -1098,6 +1097,81 @@ def transform(self, func, idx=None, tmin=None, tmax=None, copy=False):
 
         return stcs
 
+    @verbose
+    def estimate_snr(self, info, fwd, cov, verbose=None):
+        """Compute time-varying SNR in the source space
+
+        This function should only be used with source estimates with units
+        nA (i.e., MNE-like solutions, *not* dSPM or sLORETA).
+
+        Reference:
+        Goldenholz, D. M., Ahlfors, S. P., Hämäläinen, M. S., Sharon, D.,
+        Ishitobi, M., Vaina, L. M., & Stufflebeam, S. M. (2009).
+        Mapping the Signal-To-Noise-Ratios of Cortical Sources in
+        Magnetoencephalography and Electroencephalography.
+        Human Brain Mapping, 30(4), 1077–1086. doi:10.1002/hbm.20571
```

Comment:
within function docstrings, the reference should go in a separate `References` section after `Notes`. Example: https://github.com/mne-tools/mne-python/blob/master/mne/beamformer/_lcmv.py#L118-L124